app/models/comment.rb

```diff
@@ -1,4 +1,11 @@
 class Comment < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_create :update_post_comments_counte
+  after_destroy :update_post_comments_counte
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ Optional]  Here too you can use `after_save :update_comments_counter` as suggested for the post model.